### ISSUE TITLE
Workflow for testing Newsstand HPUBs.

### ISSUE THREAD

**Issue by nin9creative** [Simbul/baker#1142]
I'm just curious what the general process people are using when trying to test their final HPUBs with Baker in Newsstand mode.

I know other tools allow a nice quick and easy way to locally "preview" an issue within their viewer (Mag+, Adobe, etc) and the process with Baker is a bit more tedious.

For a project that I'm helping out with, we are using MagRocket as the backend and Amazon S3 as the .hpub issue store.  Right now the content producer is viewing the individual .html files in Safari and whatnot to test the content.  Then when they want to preview in Baker they are using a special TestFlight build that returns Issues that are not yet live on MagRocket.  

However the process isn't very smooth because there are the manual steps of building the .zip file, renaming it to .hpub, uploading to S3, refreshing the shelf on the TestFlight app, then deleting the old version of the Issue.  Clicking download to download the newly updated Issue and then opening it to test.  If a mistake is found or something else is wrong, the process starts all over - making it a bit painful.

Just curious how others there are tackling the content production and testing within Baker...

Andrew

**Comment by nicholasmartin** [comment]
When I want to preview the issue on my iPad I use a standalone baker Xcode project and just drop the issue into the books folder. 

Obviously this is fast to view the issue on the device, but doesn't test the whole baker/backend process. If it's just for checking the content then it will work (assuming your client has Xcode....)

**Comment by nin9creative** [comment]
Yeah, I was trying to avoid having to have a dependency on XCode but that's probably the easiest way to go with regards to on the fly testing of an Issue being built.  I created a special version deployed via TestFlight so that multiple folks from the client could preview it but it's not quick enough for ongoing development quick changes, fixes etc...

**Comment by JamesMowatt** [comment]
I saw a php based app not that long ago, which I have used and it worked very well. You simply dropped the unzipped folder into it's folder on a webserver and went to the URL, it allowed you to work through the magazine. I will see if I can find it again.

https://github.com/festaans/HPub-Web-Viewer
